core/history_api.php

```diff
@@ -203,7 +203,7 @@ function history_get_range_result_filter( $p_filter, $p_start_time = null, $p_en
 		return db_empty_result();
 	}
 
-	$t_select_string = 'SELECT DISTINCT {bug}.id ';
+	$t_select_string = 'SELECT {bug}.id ';
```

Comment:
Note that the filter query can return duplicated rows.
Without DISTINCT in the subquery, using a plain JOIN, DISTINCT is needed at top level select.

Looking at that query, the reported cost is slightly higher (oracle, postgres)
and execution time, in some quick tests, this is having ~10k history rows returned (the usual chunk fetched by timeline):
oracle : slightly slower than semi-join
postgres: slightly faster than semi-join
mysql (mariadb): roughly the same semi-join

However, the distinct form scales worse with result size
testing mysq, going into 1 million history rows returned:
select distinct: ~30 secs
semi-join: ~3 secs